### PR TITLE
1.8 release announcement

### PR DESCRIPTION
**Pull request by steveklabnik**
/cc @rust-lang/core

### CODE REVIEW

File: _posts/2016-04-14-Rust-1.8.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+This will print out `Count { value: 6 }`. Like the other operator traits, an
+associated type allows you to use different types on each side of the operator,
+as well. See the RFC for more details.
+
+The second feature is very small, and comes from [RFC 218]. Before Rust 1.8, a
+`struct` with no fields did not have curly braces:
+
+```rust
+struct Foo; // works
+struct Bar { }; // error
```
Comment (nikomatsakis) [inline _posts/2016-04-14-Rust-1.8.md]: Nit: no terminating semicolon here

File: _posts/2016-04-14-Rust-1.8.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+[RFC 218]: https://github.com/rust-lang/rfcs/blob/master/text/0218-empty-struct-with-braces.md
+
+The second form is no longer an error, and is equivalent to the first. This was
```
Comment (nikomatsakis) [inline _posts/2016-04-14-Rust-1.8.md]: Nit: they are not *entirely* equivalent -- e.g., in the first form, you can do `let x = Foo;` but you must do `let y = Bar { };`

File: _posts/2016-04-14-Rust-1.8.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+originally disallowed for consistency with other empty declarations, as well as
+a parsing ambiguity. However, that ambiguity is non-existent in post-1.0 Rust,
+and macro authors saw additional complexity due to needing a special-case. Also,
+users who do active development would sometimes switch between empty and
+non-empty versions of a struct, and the extra work and diffs involved was less
+than ideal. Allowing both forms is slightly less consistent, but the ergonomics
+of these two cases was deemed worth it.
```
Comment (nikomatsakis) [inline _posts/2016-04-14-Rust-1.8.md]: I feel like this is a bit more defensive than necessary. From my point-of-view, permitting `{ }` is **more** consistent, not less. That is, `struct Foo;` is a kind of shorthand for the combination of `struct Foo { }` and `const Foo: Foo = Foo { };`. The fact that we only permitted the shorthand before was a bit odd.

File: _posts/2016-04-14-Rust-1.8.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+There were a few small updates to Cargo:
+
+* [`cargo init`](https://github.com/rust-lang/cargo/pull/2081) can be used to
+  start a Cargo project in your current working directory, rather than making a
+  new subdirectory like `cargo new`.
+* `Cargo.toml` now has [keys for `-v` and `--color`](https://github.com/rust-lang/cargo/pull/2397)
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: This is actually in `.cargo/config`, not `Cargo.toml`

File: _posts/2016-04-14-Rust-1.8.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+* Various improvements to `CString`, used for FFI.
+* checked, saturated, and overflowing operations for various numeric types.
+  These aren’t counted in that ‘40’ number above, because there are a _lot_ of
+  them, but they all do the same thing.
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+There were a few small updates to Cargo:
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: Configuration via environment variables was a relatively large change, perhaps that could be mentioned here in lieu of `-v` and `--color` in `.cargo/config`?

Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: Actually https://github.com/rust-lang/cargo/pull/2328 was a pretty major feature that landing in Cargo this cycle

Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: https://github.com/rust-lang/cargo/pull/2196 (`cargo metadata`) was also a pretty major feature added this release

File: _posts/2016-04-14-Rust-1.8.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+* [`cargo init`](https://github.com/rust-lang/cargo/pull/2081) can be used to
+  start a Cargo project in your current working directory, rather than making a
+  new subdirectory like `cargo new`.
+* `Cargo.toml` now has [keys for `-v` and `--color`](https://github.com/rust-lang/cargo/pull/2397)
+
+See the release notes for the full list of changes.
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: Could this hyperlink to the notes?

File: _posts/2016-04-14-Rust-1.8.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+[now implement unwinding]: https://github.com/rust-lang/rust/pull/30448
+
+Finally, we have used `make` to build Rust for a very long time. However,
+we already have a wonderful tool for building Rust programs: Cargo. In Rust
+1.8, [initial support landed] for a new build system that’s written in Rust,
+and based on Cargo. It is not yet the default, and there is much more work to
+do. We will talk about this in release notes more once it’s completely done,
+for now, please read the GitHub issue for more details.
+
+[initial support landed]: https://github.com/rust-lang/rust/pull/30448
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: This is the same link as the MSVC one above

File: _posts/2016-04-14-Rust-1.8.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+
+#### Library stabilizations
+
+About 20 library functions and methods are now stable in 1.8. There are three
+major groups of changes: UTF-16 related string methods, various APIs related to
+time, and the various traits needed for operator overloading mentioned in the
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: I missed `std::time` the first time I read this, but it's a pretty major feature of this release, could that be highlighted a bit more?

File: _posts/2016-04-14-Rust-1.8.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+* [`cargo init`](https://github.com/rust-lang/cargo/pull/2081) can be used to
+  start a Cargo project in your current working directory, rather than making a
+  new subdirectory like `cargo new`.
+* `Cargo.toml` now has [keys for `-v` and `--color`](https://github.com/rust-lang/cargo/pull/2397)
+
+See the release notes for the full list of changes.
+
+### Contributors to 1.7
+
+We had 126 individuals contribute to 1.7. Thank you so much!
```
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: "1.7" here and the heading above

**Comment by alexcrichton** [comment]
:train2:

File: _posts/2016-04-14-Rust-1.8.md (hunk 1 above)
Comment (steveklabnik) [inline _posts/2016-04-14-Rust-1.8.md]: gah, i _always_ mess this up

File: _posts/2016-04-14-Rust-1.8.md (hunk 8 above)
Comment (steveklabnik) [inline _posts/2016-04-14-Rust-1.8.md]: I'm having a hard time coming up with a good summary, ideas?

**Comment by steveklabnik** [comment]
@alexcrichton @nikomatsakis r? I fixed almost everything, one question about how to properly summarize the time changes

File: _posts/2016-04-14-Rust-1.8.md (hunk 8 above)
Comment (alexcrichton) [inline _posts/2016-04-14-Rust-1.8.md]: Hm on rereading it's probably fine, so :+1: from me

**Comment by steveklabnik** [comment]
:shipit:

File: _posts/2016-04-14-Rust-1.8.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+Other notable improvements include:
+
+* `<[T]>::clone_from_slice()`, an efficient way to copy the data from one slice
+  and put it into another slice.
+* Various convenience methods on `Ipv4Addr` and `Ipv6Addr`, such as `is_loopback()`,
+  which returns `true` or `false` if the address is a loopback address according to
+  RFC 6890.
+* Various improvements to `CString`, used for FFI.
+* checked, saturated, and overflowing operations for various numeric types.
+  These aren’t counted in that ‘40’ number above, because there are a _lot_ of
```
Comment (SimonSapin) [inline _posts/2016-04-14-Rust-1.8.md]: Isn’t the number 20 rather than 40? Or is 40 including all the new numeric types methods?

Comment (steveklabnik) [inline _posts/2016-04-14-Rust-1.8.md]: yes, this was a typo. i've since fixed it

**Pull request merged**